src/Npgsql/Internal/Composites/Metadata/CompositeFieldInfo.cs

```diff
@@ -114,7 +114,7 @@ protected ValueTask WriteAsObject(bool async, PgConverter converter, PgWriter wr
 
     public abstract void ReadDbNull(CompositeBuilder builder);
     public abstract ValueTask Read(bool async, PgConverter converter, CompositeBuilder builder, PgReader reader, CancellationToken cancellationToken = default);
-    public abstract bool IsDbNull(PgConverter converter, object instance);
+    public abstract bool IsDbNull(PgConverter converter, object instance, ref object? writeState);
```

Comment:
Adding this just for the object[] case is a bit :vomiting_face:... But oh well

Reply:
Yeah...